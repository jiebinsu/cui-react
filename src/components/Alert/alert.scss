@import "@coopdigital/component-notification/dist/notification.css";

.coop-c-notification--alert {
  background-color: var(--color-orange-alert-light);
  border-left-color: var(--color-orange-alert);
}
