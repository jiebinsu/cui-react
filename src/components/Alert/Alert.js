import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";
import "./alert.scss";

const Alert = ({ type, heading, link, children }) => {
  const { href, text } = link;
  const classes = classNames("coop-c-notification", [
    type === "warn" && "coop-c-notification--alert",
  ]);

  return (
    <div className={classes} role="status">
      <h3 className="coop-c-notification__heading">{heading}</h3>
      <p className="coop-c-notification__p">
        {href && (
          <a href={href} className="coop-c-notification__link">
            {text}
          </a>
        )}
      </p>
      {children && <p className="coop-c-notification__p">{children}</p>}
    </div>
  );
};

Alert.defaultProps = {
  type: "info",
  link: {
    href: "",
    text: "",
  },
  children: null,
};

Alert.propTypes = {
  type: PropTypes.oneOf(["info", "warn"]),
  heading: PropTypes.string.isRequired,
  link: PropTypes.shape({
    href: PropTypes.string,
    text: PropTypes.string,
  }),
  children: PropTypes.node,
};

export default Alert;
