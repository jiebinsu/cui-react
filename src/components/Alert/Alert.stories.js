import React from "react";
import Alert from "./Alert";

export default {
  title: "Base/Alert",
  component: Alert,
};

export const Basic = (args) => <Alert {...args} />;
Basic.args = {
  heading: "Example heading",
  link: {
    href: "#",
    text: "Example link text",
  },
};

const link = { href: "#", text: "Example message" };
export const All = () => (
  <>
    <Alert type="info" heading="Info Alert" link={link} />
    <Alert type="warn" heading="Warning Alert" link={link} />
    <Alert heading="Alert with no link">Standard message</Alert>
  </>
);
