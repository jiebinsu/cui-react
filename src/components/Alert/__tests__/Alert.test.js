import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Alert from "../index";

describe("Alert", () => {
  it("should render default Alert", () => {
    const wrapper = mount(<Alert />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Alert />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should work with all variants", () => {
    const link = { href: "#", text: "Example message" };
    const Variants = () => (
      <div>
        <Alert type="info" heading="Info Alert" link={link} />
        <Alert type="warn" heading="Warning Alert" link={link} />
        <Alert heading="Alert with no link">Standard message</Alert>
      </div>
    );

    const wrapper = mount(<Variants />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Variants />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
