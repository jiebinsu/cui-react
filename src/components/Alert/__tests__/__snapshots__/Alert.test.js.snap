// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render default Alert 1`] = `
<div
  className="coop-c-notification"
  role="status"
>
  <h3
    className="coop-c-notification__heading"
  />
  <p
    className="coop-c-notification__p"
  >
    
  </p>
</div>
`;

exports[`Alert should work with all variants 1`] = `
<div>
  <div
    className="coop-c-notification"
    role="status"
  >
    <h3
      className="coop-c-notification__heading"
    >
      Info Alert
    </h3>
    <p
      className="coop-c-notification__p"
    >
      <a
        className="coop-c-notification__link"
        href="#"
      >
        Example message
      </a>
    </p>
  </div>
  <div
    className="coop-c-notification coop-c-notification--alert"
    role="status"
  >
    <h3
      className="coop-c-notification__heading"
    >
      Warning Alert
    </h3>
    <p
      className="coop-c-notification__p"
    >
      <a
        className="coop-c-notification__link"
        href="#"
      >
        Example message
      </a>
    </p>
  </div>
  <div
    className="coop-c-notification"
    role="status"
  >
    <h3
      className="coop-c-notification__heading"
    >
      Alert with no link
    </h3>
    <p
      className="coop-c-notification__p"
    >
      
    </p>
    <p
      className="coop-c-notification__p"
    >
      Standard message
    </p>
  </div>
</div>
`;
